src/appleseed/renderer/utility/dynamicspectrum.h

```diff
@@ -106,6 +108,13 @@ class DynamicSpectrum
     ValueType& operator[](const size_t i);
     const ValueType& operator[](const size_t i) const;
 
+    // Iterators.
+    iterator begin();
```

Comment:
I'm not entirely convinced that these make sense on a spectrum... If they do make sense, we should probably also add them to all our array classes (`Vector`, `Color` and `RegularSpectrum`).